pkg/provisioner/crdprovisioner.go

```diff
@@ -288,17 +297,34 @@ func (c *CrdProvisioner) DeleteVolume(ctx context.Context, volumeID string, secr
 	// return the AzVolume name to the caller as the volume ID. To make this work, we would need to implement
 	// snapshot APIs through the CRD provisioner.
 	// Replace them in all instances in this file.
+	var err error
 	lister := c.conditionWatcher.informerFactory.Disk().V1beta1().AzVolumes().Lister().AzVolumes(c.namespace)
 	azVolumeClient := c.azDiskClient.DiskV1beta1().AzVolumes(c.namespace)
 
-	volumeName, err := azureutils.GetDiskName(volumeID)
+	var volumeName string
+	volumeName, err = azureutils.GetDiskName(volumeID)
 	if err != nil {
-		klog.Errorf("Error finding volume : %v", err)
 		return nil
 	}
+
 	azVolumeName := strings.ToLower(volumeName)
 
-	waiter, err := c.conditionWatcher.newConditionWaiter(ctx, azVolumeType, azVolumeName, func(obj interface{}, objectDeleted bool) (bool, error) {
+	var azVolumeInstance *diskv1beta1.AzVolume
+	azVolumeInstance, err = lister.Get(azVolumeName)
+	ctx, workflow := azureutils.StartWorkflow(ctx, "CrdProvisioner.DeleteVolume", azureutils.GetObjectDetails(azVolumeInstance)...)
+	defer func() { workflow.Finish(err) }()
+
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			workflow.Logger().Logger.WithValues(consts.VolumeNameLabel, volumeName).Info("Deletion successful: could not find the AzVolume CRI")
```

Comment:
Hasn't the volume name already been added to the details by `azureutils.GetObjectDetails()`?

Reply:
The values need to be manually added to the logger when object is not found because object details cannot be extracted from nil objects.